contracts/src/L1/rollup/ScrollChain.sol

```
@@ -351,27 +417,88 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
+    /// @inheritdoc IScrollChain
+    /// @dev Memory layout of `_blobDataProof`:
+    /// ```text
+    /// | z       | y       | kzg_commitment | kzg_proof |
+    /// |---------|---------|----------------|-----------|
+    /// | bytes32 | bytes32 | bytes48        | bytes48   |
+    /// ```
+    function finalizeBatchWithProof4844(
+        bytes calldata _batchHeader,
+        bytes32 _prevStateRoot,
+        bytes32 _postStateRoot,
+        bytes32 _withdrawRoot,
+        bytes calldata _blobDataProof,
+        bytes calldata _aggrProof
+    ) external override OnlyProver whenNotPaused {
+        if (_prevStateRoot == bytes32(0)) revert ErrorPreviousStateRootIsZero();
+        if (_postStateRoot == bytes32(0)) revert ErrorStateRootIsZero();
 
-                    _queue.popCrossDomainMessage(_startIndex, _count, _skippedBitmap);
+        // compute batch hash and verify
+        (uint256 memPtr, bytes32 _batchHash, uint256 _batchIndex, ) = _loadBatchHeader(_batchHeader);
+        bytes32 _dataHash = BatchHeaderV1Codec.dataHash(memPtr);
+        bytes32 _blobVersionedHash = BatchHeaderV1Codec.blobVersionedHash(memPtr);
 
-                    _startIndex += 256;
-                }
-            }
+        // Calls the point evaluation precompile and verifies the output
+        {
+            (bool success, bytes memory data) = POINT_EVALUATION_PRECOMPILE_ADDR.staticcall(
+                abi.encodePacked(_blobVersionedHash, _blobDataProof)
+            );
+            // We verify that the point evaluation precompile call was successful by testing the latter 32 bytes of the
+            // response is equal to BLS_MODULUS as defined in https://eips.ethereum.org/EIPS/eip-4844#point-evaluation-precompile
+            if (!success) revert ErrorCallPointEvaluationPrecompileFailed();
+            (, uint256 result) = abi.decode(data, (uint256, uint256));
+            if (result != BLS_MODULUS) revert ErrorUnexpectedPointEvaluationPrecompileOutput();
```

What's the rationale of this check? Is it that, if the precompile is upgraded/changed in the future, then this call won't "silently pass", but instead we'll be forced to upgrade the contracts?

---

yes